### PR TITLE
Add Milestone endpoints

### PR DESCRIPTION
**Pull request by allardbrain**
**What changes does this PR introduce?**
- Moves the client logic out of `__init__.py` and into `api.py`
- Adds internal helper methods for calling requests to the `ClubhouseClient` class in `api.py`
- Adds a method for each [Milestone endpoint](https://clubhouse.io/api/rest/v3/#Milestones) offered by the Clubhouse API
- Updates all dependency versions

**Notes**
- The scope of this PR is larger than what would typically be acceptable. I set out to just add the Milestone endpoints, but made some structural changes as I discovered opportunities in the existing code. Now that I'm familiar with the code, I don't expect future PRs to be this large (or include out-of-scope changes).

**Tests**
- I need to learn how to mock requests for unit tests. In an effort to maintain momentum on this project, I've decided to focus on adding as many endpoints as I can and creating a new issue for each group of endpoints that needs testing. (For example, I'll create an issue for adding all Milestone endpoint tests.)

### CODE REVIEW

**Comment by allardbrain** [comment]
Hey there, @j-martin 👋 Do you have any interest in reviewing this PR?

File: clubhouse/__init__.py [hunk 1]
```diff
@@ -1,60 +1,3 @@
...
-        return self._request("get", *segments, **kwargs)
-
-    def post(self, *segments, **kwargs):
-        return self._request("post", *segments, **kwargs)
-
-    def put(self, *segments, **kwargs):
-        return self._request("put", *segments, **kwargs)
-
-    def delete(self, *segments, **kwargs):
-        return self._request("delete", *segments, **kwargs)
```
Comment (j-martin) [inline clubhouse/__init__.py:38]: I think there is some value keeping those around as an escape hatch and for backward compatibility.

File: clubhouse/api.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        '''Retrieve a specific Milestone.
+        https://clubhouse.io/api/rest/v3/#Get-Milestone
+
+        Example:
+            from clubhouse import ClubhouseClient
+            conn = ClubhouseClient(API_KEY)
+            conn.get_milestone(123)
+
+        Args:
+            id (int): The Milestone ID
```
Comment (j-martin) [inline clubhouse/api.py:175]: I would personally add type hints (not that I did in the previous version), although that would impact compatibility with older versions of python.

File: clubhouse/__init__.py (hunk 1 above)
Comment (allardbrain) [reply clubhouse/__init__.py:38]: You're right, I'll add those back in. Thank you.

File: clubhouse/api.py (hunk 2 above)
Comment (allardbrain) [reply clubhouse/api.py:175]: Ah, interesting. I will look into this angle, thanks!

Comment (allardbrain) [reply clubhouse/api.py:175]: Still looking into this, but resolving for now. May add this in.

**Pull request merged**